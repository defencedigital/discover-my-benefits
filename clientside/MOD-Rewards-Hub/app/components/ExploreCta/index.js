import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Row, Col } from 'reactstrap';

const ExploreCta = props => (
  <div className={`explore-cta ${props.class ? props.class : ''} `}>
    <Row>
      <Col sm="12">
        <h3 className="h2 explore-cta-title">{props.title}</h3>
      </Col>
    </Row>
    <Row>
      {props.description && (
        <>
          <Col md="12" lg="7">
            <p>{props.description}</p>
          </Col>
          <Col md="12" lg="1"></Col>
        </>
      )}
      {props.link && (
        <>
          <Col md="9" lg="4">
            <Link className="btn btn-lg btn-primary btn-wrap-xs" to={props.link}>
              {props.buttonText}
            </Link>
          </Col>
        </>
      )}
    </Row>
  </div>
);

ExploreCta.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
  link: PropTypes.string,
  buttonText: PropTypes.string,
  class: PropTypes.string,
};

export default ExploreCta;
