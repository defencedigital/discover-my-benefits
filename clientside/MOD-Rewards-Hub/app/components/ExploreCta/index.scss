.explore-cta{
  padding: 40px;
  background-color: $gray-lighter;
  color: $brand-primary;
  border-radius: 5px;
  margin-top: 60px;
  margin-bottom: 60px;
  @include media-breakpoint-down(sm) {
    padding: 20px;
  }
}


.explore-cta--pullup{
  margin-bottom: 0;
}

.explore-cta-title{
  margin-bottom: 25px;
}


.explore-cta .btn-right{
  float: right;
  // display: block;
  vertical-align: bottom;
  max-width: 100%;

  @media screen and (max-width: $breakpoint-lg) {
    float: none;
  }
}

