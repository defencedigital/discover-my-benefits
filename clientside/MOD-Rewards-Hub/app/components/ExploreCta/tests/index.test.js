import React from 'react';
import { fromJS } from 'immutable';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import ExploreCta from '../index';

Enzyme.configure({ adapter: new Adapter() });

const mockedProps = fromJS({
  title: 'title',
  items: ['one', 'two'],
  link: '/link',
});

describe('ExploreCta component', () => {
  it('matches the snapshot', () => {
    const html = shallow(<ExploreCta {...mockedProps.toJS()} />);
    expect(html).toMatchSnapshot();
  });
});
