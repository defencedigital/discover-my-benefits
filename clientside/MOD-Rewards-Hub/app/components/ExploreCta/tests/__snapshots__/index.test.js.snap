// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExploreCta component matches the snapshot 1`] = `ShallowWrapper {}`;
